app/models/epp/contact.rb

```diff
@@ -16,6 +16,26 @@ def manage_permissions
     throw(:abort)
   end
 
+  validate :validate_birthday_ident
+
+  def validate_birthday_ident
+    return unless Depp::Contact::SELECTION_TYPES[2].include?(ident_type)
+
+    if (Date.parse(ident) rescue ArgumentError) == ArgumentError
+      add_epp_error('2308', nil, nil, I18n.t('epp.contacts.errors.valid_ident_date_format'))
+      @error = true
+      return
+    end
+
+    contact_ident_date = Date.parse(ident)
+    valid_time_range = (Date.today - 125.years)...(Date.tomorrow - 18.years)
```

Comment:
Let's save it in some constant or Value Object.
And why 18 years ago? Do we have any legal regulations on that?

Reply:
I am not aware of any legal regulations regarding this. This is just my guess. I'll ask Timo about it.